### ISSUE TITLE
About MapTR V2

### ISSUE THREAD

**Issue by xpyqiubai** [boschresearch/PseudoMapTrainer#1]
Thank you for your excellent work. This method is inspiring for cases where data annotation is not possible. Table 2 in the text mainly compares the MapTRv2 method. However, it seems that only the MapVR-related content is included in the code. Could you please share the training code related to MapTRv2? Thank you very much! 
In addition, for the train model of segmentation on Mapillary V2. Can the trained weights be shared? If possible, thank you very much! My email address is xpydgqb@gmail.com.

**Comment by christianll9** [comment]
Hi, thanks for your interest!

We used the MapVR repo as a base (see ceaa00c) and then combined it with the advancements of MapTRv2 (see cc4b989). You can run the MapTRv2 training with GT and pseudo-labels using the configs in [`./MapVR/projects/configs/maptrv2`](https://github.com/boschresearch/PseudoMapTrainer/blob/main/MapVR/projects/configs/maptrv2).

Unfortunately, I cannot publish the weights, but we provide the full code for training the segmentation model (see [1.1.5](https://github.com/boschresearch/PseudoMapTrainer?tab=readme-ov-file#11-get-pv-segmentation-labels-mask2former) of our tutorial).

**Comment by xpyqiubai** [comment]
> Hi, thanks for your interest!
> 
> We used the MapVR repo as a base (see [ceaa00c](https://github.com/boschresearch/PseudoMapTrainer/commit/ceaa00c72aef208f5293486e30debef936caf0aa)) and then combined it with the advancements of MapTRv2 (see [cc4b989](https://github.com/boschresearch/PseudoMapTrainer/commit/cc4b989d7765d918e31bbba86bec8dccf3c7fbc5)). You can run the MapTRv2 training with GT and pseudo-labels using the configs in [`./MapVR/projects/configs/maptrv2`](https://github.com/boschresearch/PseudoMapTrainer/blob/main/MapVR/projects/configs/maptrv2).
> 
> Unfortunately, I cannot publish the weights, but we provide the full code for training the segmentation model (see [1.1.5](https://github.com/boschresearch/PseudoMapTrainer?tab=readme-ov-file#11-get-pv-segmentation-labels-mask2former) of our tutorial).

Thanks a lot! It seems that since you have optimized MapVR based on MapTRv2, theoretically, following the steps in the Readme, I should be able to achieve better performance (compared to MapTRv2). I would like to ask again - approximately how long does it take to train the segmentation model? Thank you!

**Comment by christianll9** [comment]
Yes, but the combination of MapVR and MapTRv2 is only enabled for the `pmt_...` (=PseudoMapTrainer) configs. Using the `maptrv2_...` configs would start a standard MapTRv2 training.

Training the segmentation network took around one week using two A100s.

**Comment by xpyqiubai** [comment]
> Yes, but the combination of MapVR and MapTRv2 is only enabled for the `pmt_...` (=PseudoMapTrainer) configs. Using the `maptrv2_...` configs would start a standard MapTRv2 training.
> 
> Training the segmentation network took around one week using two A100s.

May I ask which version of your A100, the 40g or the 80g? Even when I reduced the batch_size to 40g, A100, the error "cuda memory out" still occurred.

**Comment by christianll9** [comment]
Hi @xpyqiubai, we used two A100s with 80GB each, a batch size of 8, and a lr of 0.00008.

**Comment by xpyqiubai** [comment]
Hello, the performance indicators of the mapillary_vistas_v2_sem_seg_val model I trained are as follows:

<img width="923" height="78" alt="Image" src="https://github.com/user-attachments/assets/55b8ebc1-53b8-4388-a724-e0521b564c75" />

Would you like to confirm with me whether this indicator is correct?

**Comment by christianll9** [comment]
Hi @xpyqiubai,

I do not have the evaluation results for the segmentation model recorded. However, a relatively quick way to evaluate your model would be to generate single-trip pseudo-labels for the validation split only and then compare the results with those in Table 1 in our paper. To do so:
1. Set `splits: ["val"]` in [`RoGS/configs/nusc/single_trip.yaml#L21`](https://github.com/boschresearch/PseudoMapTrainer/blob/09300d5bfa56f7282605b3c36cc7d7d7cf2395a2/RoGS/configs/nusc/single_trip.yaml#L21)
2. When preprocessing the labels according to the tutorial in [2.2)](https://github.com/boschresearch/PseudoMapTrainer/tree/main?tab=readme-ov-file#2-training-the-online-mapping-model-mapvr), add the flag `--only-val-split`
3. You can skip the training in 2.5) and evaluate the pseudo-labels directly, as described in the [evaluation tutorial](https://github.com/boschresearch/PseudoMapTrainer/tree/main?tab=readme-ov-file#evaluation).

I hope this helps.

**Comment by xpyqiubai** [comment]
> Hi [@xpyqiubai](https://github.com/xpyqiubai),
> 
> I do not have the evaluation results for the segmentation model recorded. However, a relatively quick way to evaluate your model would be to generate single-trip pseudo-labels for the validation split only and then compare the results with those in Table 1 in our paper. To do so:
> 
> 1. Set `splits: ["val"]` in [`RoGS/configs/nusc/single_trip.yaml#L21`](https://github.com/boschresearch/PseudoMapTrainer/blob/09300d5bfa56f7282605b3c36cc7d7d7cf2395a2/RoGS/configs/nusc/single_trip.yaml#L21)
> 2. When preprocessing the labels according to the tutorial in [2.2)](https://github.com/boschresearch/PseudoMapTrainer/tree/main?tab=readme-ov-file#2-training-the-online-mapping-model-mapvr), add the flag `--only-val-split`
> 3. You can skip the training in 2.5) and evaluate the pseudo-labels directly, as described in the [evaluation tutorial](https://github.com/boschresearch/PseudoMapTrainer/tree/main?tab=readme-ov-file#evaluation).
> 
> I hope this helps.

Thank you. I will carry out the evaluation according to your suggestions.
